Modules/Loadable/Tables/Logic/vtkSlicerTablesLogic.cxx

```diff
@@ -68,27 +72,75 @@ ::AddTable(const char* fileName, const char* name)
     return 0;
     }
 
-  // Storage node
-  vtkNew<vtkMRMLTableStorageNode> tableStorageNode;
-  tableStorageNode->SetFileName(fileName);
-  this->GetMRMLScene()->AddNode(tableStorageNode.GetPointer());
-
   // Storable node
-  vtkNew<vtkMRMLTableNode> tableNode;
-  if (name)
-    {
-    tableNode->SetName(name);
-    }
-  this->GetMRMLScene()->AddNode(tableNode.GetPointer());
+  vtkMRMLTableNode *tableNode;
 
-  // Read
-  int res = tableStorageNode->ReadData(tableNode.GetPointer());
-  if (res == 0) // failed to read
+  if (std::string(fileName).find(".db") != std::string::npos)
     {
-    vtkErrorMacro("vtkSlicerTablesLogic::AddTable failed: failed to read data from file: "<<fileName);
-    this->GetMRMLScene()->RemoveNode(tableStorageNode.GetPointer());
-    this->GetMRMLScene()->RemoveNode(tableNode.GetPointer());
-    return 0;
+    // SQLite
+    std::string dbname = std::string("sqlite://") + std::string(fileName);
```

Comment:
Add a static method to the storage node that returns a list of tables from a filename to remove repeating very low-level details, such as creating dbname from filename, opening database, etc.